docs/source/overview/model-inference.rst

```diff
@@ -0,0 +1,77 @@
+
+1. Projection
+
+   The most common usecases are model inference in projections. For example, we can use the `MnistImageClassifier <https://github.com/georgia-tech-db/evadb/blob/staging/evadb/functions/mnist_image_classifier.py>`_ to identify numbers from the `MINST <https://www.dropbox.com/s/yxljxz6zxoqu54v/mnist.mp4>`_ video. 
+
+.. code-block:: sql
+
+   SELECT MnistImageClassifier(data).label FROM minst_vid;
+
+2. Selection
+
+   Another common usecases are model inference in selections. In the below example, we use ``TextSummarizer`` and ``TextClassifier`` from :ref:`HuggingFace<hf>` to summarize the negative food reviews.
+
+.. code-block:: sql
+
+   SELECT TextSummarizer(data)
+   FROM food_reviews
+   WHERE TextClassifier(data).label = 'NEGATIVE';
+
+EvaDB also provides specialized array operators to construct queries. Go over built-in utility operators and functions for all of them. Below is an example of ``CONTAIN``:
+
+.. code-block:: sql
+
+   SELECT id FROM camera_videos 
+   WHERE ObjectDetector(data).labels @> ['person', 'car'];
+
+3. Lateral Join
+
+   In EvaDB, we can also use models in joins.
+   The most powerful usecase is lateral join combined with ``UNNEST``, which is very helpful to flatten the output from `one-to-many` models.
+   The key idea here is a model could give multiple outputs (e.g., bounding box) stored in an array. This syntax is used to unroll elements from the array into multiple rows.
+   Typical examples are `face detectors <https://github.com/georgia-tech-db/evadb/blob/staging/evadb/functions/face_detector.py>`_ and `object detectors <https://github.com/georgia-tech-db/evadb/blob/staging/evadb/functions/fastrcnn_object_detector.py>`_. 
+   In the below example, we use `emotion detector <https://github.com/georgia-tech-db/evadb/blob/staging/evadb/functions/emotion_detector.py>_` to detect emotions from faces in the movie, where a single scene can contain multiple faces. 
+   
+.. code-block:: sql
+   
+   SELECT EmotionDetector(Crop(data, bbox))
+   FROM movie
+   LATERAL JOIN UNNEST(FaceDetector(data)) AS Face(bbox, conf);
+
+4. Aggregate Functions
+
+   Models or user-defined functions can also act as aggregate functions for ``GROUP BY``. Typical usecases are action recognition. 
+
+.. code-block:: sql
+
+   SELECT ASLActionRecognition(SEGMENT(data)) 
+   FROM ASL_ACTIONS 
+   SAMPLE 5 
+   GROUP BY '16 frames';
+
+5. Order By
+   
+   Models (typically feature extractors) can also be used in the ``ORDER BY`` for embedding-based similarity search. EvaDB also has index support to facilitate this type of queries. In the below example, we use the `SentenceFeatureExtractor <https://github.com/georgia-tech-db/evadb/blob/staging/evadb/functions/sentence_feature_extractor.py>`_ to answer the question `When was the NATO created` from a collection of pdfs as the knowledge base.
```

Comment:
Models (typically feature extractors) can be used in the ``ORDER BY`` for embedding-based similarity search. EvaDB provides index support to facilitate such usecases. <add link to creating index>. 

replace: answer to question -> find relevant context. 
Also add the image similarity query and link to the usecase.